pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -26,11 +23,65 @@ type notificationChannel struct {
 }
 
 // channelsPerOrg maps notification channels per organisation
-type channelsPerOrg map[int64]map[interface{}]*notificationChannel
+type channelsPerOrg map[int64][]*notificationChannel
 
 // channelMap maps notification channels per organisation
 type defaultChannelsPerOrg map[int64][]*notificationChannel
 
+// uidOrID for both uid and ID, primarily used for mapping legacy channel to migrated receiver.
+type uidOrID interface{}
+
+// setupAlertmanagerConfigs starts channel and alert migrations. This creates Alertmanager configs as well as migrates receivers and creates the initial root-level route.
+// Returns per org maps containing: The alertmanager config, all migrated receivers, all migrated default receivers.
+func (m *migration) setupAlertmanagerConfigs() (amConfigsPerOrg, map[int64]map[uidOrID]*PostableApiReceiver, map[int64]map[string]struct{}, error) {
+	// allChannels: channelUID -> channelConfig
+	allChannelsPerOrg, defaultChannelsPerOrg, err := m.getNotificationChannelMap()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	amConfigPerOrg := make(amConfigsPerOrg, len(allChannelsPerOrg))
+	receiversPerOrg := make(map[int64]map[uidOrID]*PostableApiReceiver, len(allChannelsPerOrg))
+	defaultReceiversPerOrg := make(map[int64]map[string]struct{}, len(allChannelsPerOrg))
+	for orgID := range allChannelsPerOrg {
+		amConfig := &PostableUserConfig{
+			AlertmanagerConfig: PostableApiAlertingConfig{
+				Receivers: make([]*PostableApiReceiver, 0),
+			},
+		}
+		amConfigPerOrg[orgID] = amConfig
+
+		// Create all newly migrated receivers from legacy notification channels.
+		receiversMap, receivers, err := m.createReceivers(allChannelsPerOrg[orgID])
```

Comment:
you can iterate through the map and read key and value i.e `for orgID, orgChannels := range allChannelsPerOrg` and just use the variable here.